src/components/lava-coder/lava-coder-view.tsx

```diff
@@ -33,13 +36,42 @@ export function LavaCoderView({ width, height, margin }: IProps) {
         ? IonWorldImageryStyle.AERIAL_WITH_LABELS
         : IonWorldImageryStyle.AERIAL;
       createWorldImageryAsync({ style }).then((imageryProvider) => {
-        const imageryLayer = new ImageryLayer(imageryProvider);
-        widget.imageryLayers.removeAll();
-        widget.imageryLayers.add(imageryLayer);
+        // Remove the old base layer
+        const oldBaseLayer = widget.imageryLayers.get(0);
+        if (oldBaseLayer) {
+          widget.imageryLayers.remove(oldBaseLayer);
+        }
+        const newBaseLayer = new ImageryLayer(imageryProvider);
+        // Add the new base layer at the bottom of the layer stack
+        widget.imageryLayers.add(newBaseLayer, 0);
       });
     }
   }, [showLabels, widget]);
 
+  // Load the elevation data
+  useEffect(() => {
+    if (lavaSimulation.raster) return;
+
+    const rasterWorker = new RasterWorker();
+    rasterWorker.onmessage = (e) => {
+      try {
+        const { status } = e.data;
+        if (status === "rasterParsed") {
+          lavaSimulation.setRaster(e.data.raster);
+          rasterWorker.terminate();
+        }
+      } catch (error) {
+        console.error("Error handling worker message:", error, e);
+      }
+    };
+
+    rasterWorker.postMessage({ type: "start" });
+
+    return () => {
+      rasterWorker.terminate();
+    };
+  }, []);
```

Comment:
I would move this into its own custom hook, something like `useElevationData()` and then only the return value is needed here. In an upcoming PR I've done something similar for `useHazardZones()` and `useTerrainProvider()`.